### ISSUE TITLE
dartdoc outputs `unresolved export uri` error

### ISSUE THREAD

**Issue by mhmdanas** [dart-lang/dartdoc#2143]
Output of `flutter doctor`:
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, v1.12.13+hotfix.7, on Mac OS X 10.15.3 19D76, locale en)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[!] Android Studio (not installed)
[✓] IntelliJ IDEA Community Edition (version 2019.3.1)
[✓] VS Code (version 1.41.1)
[!] Connected device
    ! No devices available

! Doctor found issues in 2 categories.
```

Part of output:
```
$ flutter pub global run dartdoc
...
error: unresolved export uri: "dart:html_common", from dart-html: (file:///Users/mhmdanas/flutter/bin/cache/pkg/sky_engine/lib/html/html_dart2js.dart:17:9)
...
found 245 warnings and 1 error
Documented 15 public libraries in 165.6 seconds

dartdoc failed: dartdoc encountered 1 errors while processing..
pub finished with exit code 1
```

**Note**: The `doc` directory is still created successfully.

**Comment by jcollins-g** [comment]
@mhmdanas Is there anything you can share about the package you are documenting?

**Comment by jcollins-g** [comment]
Triaging as "P1" pending more information.

**Comment by mhmdanas** [comment]
I just created a new flutter project with `flutter create` and then ran `dartdoc`.

**Comment by kawa89** [comment]
Same here with my library https://pub.dev/packages/tweet_ui#-analysis-tab-
I got -10 points in Maintenance :-(

**Comment by shah-xad** [comment]
Like @kawa89 same with my library https://pub.dev/packages/flutter_tex#-analysis-tab-, I also got -10 points in maintenance.

**Comment by jcollins-g** [comment]
Can reproduce at head flutter as well with latest dartdoc.  Looking.

**Comment by jcollins-g** [comment]
This seems to be due to some missing URI mappings in embedder.yaml.  How Flutter works without analysis errors despite the missing mappings is curious.

**Comment by jcollins-g** [comment]
The root cause is that `dart:html_common` and its dependency, `dart:_metadata`, are not part of sky_engine's embedder mappings.   Attempting to use the symbol 'promiseToFuture' after importing 'dart:html' will show the bug in a Flutter package:

```
jcollins-macbookpro2:createfresh jcollins$ ../flutter/bin/flutter analyze
Analyzing createfresh...                                                

   info • Avoid using web-only libraries outside Flutter web plugin packages • lib/main.dart:2:1 • avoid_web_libraries_in_flutter
  error • Undefined name 'promiseToFuture' • lib/main.dart:5:9 • undefined_identifier

2 issues found. (ran in 2.8s)
jcollins-macbookpro2:createfresh jcollins$
```

This proves that the lack of html_common is a real problem, but doesn't explain why the analyzer doesn't provide a better message here.   There's probably a bug somewhere in the embedded SDK support that doesn't properly detect this situation.

The fix will need to proceed in two stages:  first, add the `.gni` data to the SDK repository, and then update the build in flutter/engine to properly copy the SDK files into sky_engine and correct the URI mapping.

**Comment by jcollins-g** [comment]
first stage of fix:  https://dart-review.googlesource.com/c/sdk/+/143020

**Comment by jcollins-g** [comment]
The fix for #1949 eliminates this bug as well and will go out in the next version.  There is still the general issue of sky_engine reexporting dart libraries with missing mappings, but I'll leave that question to flutter./flutter#54793 as it's no longer creating an acute problem for dartdoc.
